include/aikido/control/BarrettFingerKinematicSimulationPositionCommandExecutor.hpp

```diff
@@ -8,67 +8,74 @@
+    ::dart::dynamics::ChainPtr finger, size_t proximal, size_t distal,
+    ::dart::collision::CollisionDetectorPtr collisionDetector = nullptr,
+    ::dart::collision::CollisionGroupPtr collideWith = nullptr,
+    ::dart::collision::CollisionOption collisionOptions
       = ::dart::collision::CollisionOption(false, 1));
 
   /// Sets variables to move the finger joints.
-  /// proximal dof moves to _goalPosition, joint limit, or until collision.
+  /// proximal dof moves to goalPosition, joint limit, or until collision.
   /// Step method should be called multiple times until future returns.
-  /// Distal dof follows with mimic ratio. 
-  /// \param _goalPosition Desired angle of proximal joint.
-  /// \param _collideWith CollisionGroup to check collision with fingers.
+  /// Distal dof follows with mimic ratio.
+  /// \param goalPosition Desired angle of proximal joint.
   /// \return future Becomes available when the execution completes.
-  std::future<void> execute(double _goalPosition,
-    ::dart::collision::CollisionGroupPtr _collideWith);
+  std::future<void> execute(double goalPosition);
 
-  /// Returns mimic ratio, i.e. how much the distal joint moves relative to 
+  /// Returns mimic ratio, i.e. how much the distal joint moves relative to
   /// the proximal joint. The joint movements follow
   /// this ratio only when both joints are moving.
   /// \return mimic ratio.
-  static double getMimicRatio();
+  constexpr static double getMimicRatio() { return kMimicRatio; }
 
-  /// Moves the joints of the finger by dofVelocity*_timeSIncePreviousCall
+  /// Moves the joints of the finger by dofVelocity*timeSincePreviousCall
   /// until execute's goalPosition by primary dof or collision is detected.
-  /// If proximal link is in collision, distal link moves until 
+  /// If proximal link is in collision, distal link moves until
   /// mimicRatio*goalPosition. If distal link is in collision, execution stops.
   /// If multiple threads are accessing this function or skeleton associated
   /// with this executor, it is necessary to lock the skeleton before
   /// calling this method.
-  /// \param _timeSincePreviousCall Time since previous call. 
-  void step(double _timeSincePreviousCall);
+  /// \param[in] timeSincePreviousCall Time interval to take.
+  void step(const std::chrono::milliseconds& timeSincePreviousCall);
+
+  /// Resets CollisionGroup to check collision with fingers.
+  /// \param _collideWith CollisionGroup to check collision with fingers.
+  /// \return false if fails to change collideWith (during execution).
+  bool setCollideWith(::dart::collision::CollisionGroupPtr collideWith);
 
-private: 
-  constexpr static double kMimicRatio = 0.333; 
-  constexpr static double kProximalVelocity = 0.01;
-  constexpr static double kDistalVelocity = kProximalVelocity*kMimicRatio;
+private:
+  constexpr static double kMimicRatio = 0.333;
+  constexpr static double kProximalSpeed = 1;
```

Comment:
Hmm, it seems like making this change would require `kProximalSpeed` and `kDistalSpeed` to not be static, right? I'm not sure why they're static now and whether that's necessary though.